test/e2e/performanceprofile/functests/8_performance_workloadhints/workloadhints.go

```diff
@@ -617,14 +617,34 @@ var _ = Describe("[rfe_id:49062][workloadHints] Telco friendly workload specific
 					HighPowerConsumption:  pointer.Bool(true),
 					RealTime:              pointer.Bool(true),
 				}
-				EventuallyWithOffset(1, func() string {
-					err := testclient.ControlPlaneClient.Update(context.TODO(), profile)
-					if err != nil {
-						statusErr, _ := err.(*errors.StatusError)
-						return statusErr.Status().Message
-					}
-					return fmt.Sprint("Profile applied successfully")
-				}, time.Minute, 5*time.Second).Should(ContainSubstring("HighPowerConsumption and PerPodPowerManagement can not be both enabled"))
+				if hypershift.IsHypershiftCluster() {
+					hostedClusterName, err := hypershift.GetHostedClusterName()
+					np, err := nodepools.GetByClusterName(ctx, testclient.ControlPlaneClient, hostedClusterName)
+					Expect(err).ToNot(HaveOccurred())
+					EventuallyWithOffset(1, func() string {
+						err := testclient.ControlPlaneClient.Update(context.TODO(), profile)
+						if err == nil {
+							reason := "ValidationFailed"
+							messages, err := nodepools.NodePoolStatusMessages(ctx, testclient.ControlPlaneClient, np.Name, np.Namespace, reason)
+							if err != nil {
+								statusErr, _ := err.(*errors.StatusError)
+								return statusErr.Status().Message
+							}
+							return messages[0]
+						}
+						return ""
+					}, time.Minute, 5*time.Second).Should(ContainSubstring("HighPowerConsumption and PerPodPowerManagement can not be both enabled"))
+				} else {
```

Comment:
I don't understand what are we checking in the `else` block.
At glance it looks like we're just validating that the update completed successfully.

We have `UpdateWithRetry` for that:
https://github.com/openshift/cluster-node-tuning-operator/blob/a4cac8f9ebfdd2b5043f6957cba6233eb2f2d600/test/e2e/performanceprofile/functests/utils/profiles/profiles.go#L101

Reply:
In the else block we are checking that in the  standard openshift, if enabling PerPodPowerManagement and HighPowerConsumption at the same time should fail. It doesn't allow you to commit the change in the profile. 

where as in Hypershift, since it's configmap it allows the update and throws the validation in the nodepool status.